Updates the What is Kubernetes page

shavidissa opened the pull request:
Fixes https://github.com/kubernetes/website/issues/14306 and fixes https://github.com/kubernetes/website/issues/12333. The content is work in progress and will be updated once all the comments are received.

shavidissa commented:
/assign @juandiegopalomino 
/assign @steveperry-53 
/assign @Bradamant3

shavidissa commented:
Preview: https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/

sftim commented:
@shavidissa looking at this I can't easily tell what has changed

To make that more obvious you could make two commits:
- one commit that changes the wrapping and doesn't change the the rendered page
- another commit that leaves the line wrapping as-is and changes the content

Does that make sense?

tengqm commented:
I'm against the change to line wrappings. The existing wrapping works well and it is the recommended way for writing paragraphs. Merging multiple lines into a single line will make future revisions difficult to review. Ideally, we should wrap long paragraphs at a fixed column width (say 120 characters).

juandiegopalomino commented on content/en/docs/concepts/overview/what-is-kubernetes.md:
```diff
@@ -15,196 +15,71 @@ This page is an overview of Kubernetes.
+That's how Kubernetes comes to the rescue! Kubernetes provides you with a framework to run distributed system resiliently. It takes care of your scaling requirements, failover, deployment patterns, and more. For example, Kubernetes can easily manage a canary deployment for your system.
+
+Kubernetes provides you with:
+* Service discovery and load balancing
```
list not rendering well: https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#why-do-i-need-kubernetes-and-what-can-it-do

shavidissa commented:
> It's looking good so far, but the explanation of "what it is" is a bit thin-- we have a lot context of where it came from and what it's not, but not much about the identity itself.

Good point! I will add more content in.

shavidissa commented:
@juandiegopalomino: I added more content in. Can you take a look and see?

Rajakavitha1 commented:
@shavidissa This looks good!!!!
my two cents: For https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#what-kubernetes-is-not

We could emphasis about what Kubernetes supports rather than what it does not.......
Becasue, though we mention that Kubernetes does not.........we are also mentioning about the alternatives or the options that Kubernetes provides .....I would do thick on points that Kubernetes supports. 
I am not sure if we could change the title to " How Kubernetes supports common features in PaaS offerings" or something similar!!!!!

steveperry-53 commented on content/en/docs/concepts/overview/what-is-kubernetes.md:
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
+
+Each VM is a full machine running all the components, including its own operating system, on top of the virtualized hardware.
+
+**Container deployment era:** Containers are similar to VMs, but they have relaxed isolation properties to share the Operating System (OS) among the applications. Therefore, containers are considered lightweight. Similar to a VM, it has its own filesystem, CPU, memory, process space, and more. As they are decoupled from the underlying infrastructure, they are portable across clouds and OS distrib ...
```
"Similar to a VM, it ..."
Needs a change to something like:
"Similar to VMs, they ..."
"Similar to a VM, a container ..."

steveperry-53 commented on content/en/docs/concepts/overview/what-is-kubernetes.md:
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
-has enabled a number of other systems to build atop Kubernetes.
+Containers are a good way to bundle and run your applications. In a production environment, you need to manage the containers that run the applications and ensure that there is no downtime. For example, if a container goes down, another container needs to restart. Wouldn't it be easier if this behavior was handled by a system?
+
+That's how Kubernetes comes to the rescue! Kubernetes provides you with a framework to run distributed system resiliently. It takes care of your scaling requirements, failover, deployment patterns, and more. For example, Kubernetes can easily manage a canary deployment for your system.
```
run distributed systems

steveperry-53 commented on content/en/docs/concepts/overview/what-is-kubernetes.md:
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
+Kubernetes provides you with:
+
+* **Service discovery and load balancing**  
+Kubernetes can expose containers using the DNS name or using their own IP address. If traffic to a container is high, Kubernetes is able to load balance and distribute the network traffic so that the deployment is stable.
```
expose a container using its DNS name or using its IP address

steveperry-53 commented on content/en/docs/concepts/overview/what-is-kubernetes.md:
```diff
@@ -15,196 +15,77 @@ This page is an overview of Kubernetes.
+* **Self-healing**  
+Kubernetes restarts containers that fail, replaces containers, kills containers that don’t respond to your user-defined health check, and doesn’t advertise them to clients until they are ready to serve.
+* **Secret and configuration management**  
+Kubernetes lets you store and manage sensitive information, such as passwords, OAuth tokens, and ssh keys. You can deploy and update secrets and application configuration without rebuilding your image, and without exposing secrets in your stack configuration.
```
Suggestion:
without rebuilding your container images

steveperry-53 commented:
@shavidissa This looks good. I left just a few minor comments.

Rajakavitha1 commented:
Hi,

There a good point that @sftim makes in : https://kubernetes.slack.com/archives/CGW9JHPJN/p1559665692002800

Hope this helps!!!!!

Rajie

shavidissa commented:
> @shavidissa This looks good. I left just a few minor comments.

Thanks a lot for the review, @steveperry-53!

shavidissa commented:
> Hi,
> 
> There a good point that @sftim makes in : https://kubernetes.slack.com/archives/CGW9JHPJN/p1559665692002800
> 
> Hope this helps!!!!!
> 
> Rajie

Hi Rajie,

Thanks a lot! Good point there. I replied on Slack and will work to add this info to the Kubernetes components section.

shavidissa commented:
> @shavidissa This looks good!!!!
> my two cents: For https://deploy-preview-14657--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/what-is-kubernetes/#what-kubernetes-is-not
> 
> We could emphasis about what Kubernetes supports rather than what it does not.......
> Becasue, though we mention that Kubernetes does not.........we are also mentioning about the alternatives or the options that Kubernetes provides .....I would do thick on points that Kubernetes supports.
> I am not sure if we could change the title to " How Kubernetes supports common features in PaaS offerings" or something similar!!!!!

Good suggestion, @Rajakavitha1! Let me rename that heading .. that makes a lot of sense.

shavidissa commented:
Hi @steveperry-53  and @juandiegopalomino,

Addressed your reviews and updated the content. Will you be able to check and give an lgtm if it is good to go, please?

shavidissa commented:
/assign @zacharysarah

shavidissa commented:
Hi @steveperry-53,
If this looks ok, will you be able to _lgtm_ and _approve_, please? :)

steveperry-53 commented:
/approve
/lgtm
